state_bucket/main.tf

```diff
@@ -84,25 +124,24 @@ resource "aws_s3_bucket" "s3-logs" {
     prefix = "/"
 
     transition {
-      days          = 30
-      storage_class = "STANDARD_IA"
+      storage_class = "INTELLIGENT_TIERING"
     }
-
-    transition {
-      days          = 365
-      storage_class = "GLACIER"
+    noncurrent_version_transition {
+      storage_class = "INTELLIGENT_TIERING"
     }
-
     expiration {
       # 5 years
       days = 1825
     }
+    noncurrent_version_expiration {
+      days = 1825
+    }
   }
 
   server_side_encryption_configuration {
     rule {
       apply_server_side_encryption_by_default {
-        sse_algorithm = "aws:kms"
+        sse_algorithm = "AES256"
```

Comment:
Did I hear you or someone say the reason we can't use `aws:kms` here is because with S3 it uses the parent org's KMS key, not one that our org/account has access to as with the other resources (state bucket, etc) that use `aws:kms`?

Reply:
The logs are being written by an AWS service that runs in account that AWS owns.  If you set the encryption to be aws:kms, it will use the s3 encryption key from the AWS service account.  There is no way to get access to that key so you will not be able to read the log files.